packages/astro/src/core/build/graph.ts

```diff
@@ -26,11 +26,11 @@ export function* walkParentInfos(
 export function* getTopLevelPages(
 	id: string,
 	ctx: { getModuleInfo: GetModuleInfo }
-): Generator<string, void, unknown> {
+): Generator<ModuleInfo, void, unknown> {
 	for (const info of walkParentInfos(id, ctx)) {
 		const importers = (info?.importers || []).concat(info?.dynamicImporters || []);
-		if (importers.length <= 2 && importers[0] === resolvedPagesVirtualModuleId) {
-			yield info.id;
+		if (importers[0] === resolvedPagesVirtualModuleId) {
```

Comment:
@bholmesdev and I also ran into this while pairing on https://github.com/withastro/astro/pull/4137. 

I'm not sure we can always guarantee that the _first_ import is `resolvedPagesVirtualModuleId`... Would searching all importers be safer?

```suggestion
		if (importers.some((importer) => importer === resolvedPagesVirtualModuleId)) {
			yield info;
		}
```

Reply:
Did some research and [this page](https://github.com/rollup/rollup/blob/master/docs/05-plugin-development.md#thisgetmoduleinfo) is the best description of what `importers` is, and it sounds like they are added in order they are discovered. Since the virtual module is the top-level `input` it should always be the first entry in the array. I think it's safe to make the code simplified and skip the search. I'd rather know *why* it's not the first, if that were to ever occur.